Plugin.php

```diff
@@ -366,4 +371,48 @@ public function translateRawPlural($string, $count = 0, $params = [], $locale =
     {
         return Lang::choice(Message::transRaw($string, $params, $locale), $count, $params);
     }
+
+    public function extendWinterSitemap()
+    {
+        Event::listen('winter.sitemap.processMenuItems', function ($item, $url, $theme, $apiResult) {
+            switch ($item->type) {
+                case 'cms-page':
+                    return Classes\MLCmsPage::resolveMenuItem($item, $url, $theme);
+                case 'blog-category':
+                case 'all-blog-categories':
+                    return Classes\MLBlogCategoryModel::resolveMenuItem($item, $url, $theme);
+                case 'blog-post':
+                case 'category-blog-posts':
+                case 'all-blog-posts':
+                    return Classes\MLBlogPostModel::resolveMenuItem($item, $url, $theme);
+                default:
+                    return false;
+            }
+        });
+
+        Event::listen('winter.sitemap.makeUrlSet', function ($definition, $xml, $urlSet) {
+            $botsRegex = '/(Apple|Bing|Google|DuckDuck|Yandex|Exa|Face)[bB]ot|Baiduspider|Slurp|Sogou/';
+            if (preg_match($botsRegex, Request::server('HTTP_USER_AGENT', ''))) {
+                $nsUrl = 'http://www.w3.org/1999/xhtml';
+            } else {
+                // hack to force browser to properly render the XML sitemap
+                $nsUrl = 'xmlns:xhtml-namespace-definition-URL-here';
+            }
```

Comment:
if this hack is just for debugging purpose, would not be safer to test if `debug` exists in query string, avoiding potential problem with non-listed bots by default?

Reply:
Fixed.

Reply:
Just some thinking;
How can the bots read my sitemap on staging environment? Debug mode will be active

I don't know how these work; but things like SiteGuru and ScreamingFrog should still work on staging environments?